flight/targets/DTFc/board-info/board_hw_defs.c

```diff
@@ -0,0 +1,942 @@
+        },
+        .pin_source = GPIO_PinSource3,
+    },
+    .miso = {
+        .gpio = GPIOB,
+        .init = {
+            .GPIO_Pin = GPIO_Pin_4,
+            .GPIO_Speed = GPIO_Speed_50MHz,
+            .GPIO_Mode = GPIO_Mode_AF,
+            .GPIO_OType = GPIO_OType_PP,
+            .GPIO_PuPd = GPIO_PuPd_NOPULL
+        },
+        .pin_source = GPIO_PinSource4,
+    },
+    .mosi = {
+        .gpio = GPIOB,
+        .init = {
+            .GPIO_Pin = GPIO_Pin_5,
+            .GPIO_Speed = GPIO_Speed_50MHz,
+            .GPIO_Mode = GPIO_Mode_AF,
+            .GPIO_OType = GPIO_OType_PP,
+            .GPIO_PuPd = GPIO_PuPd_NOPULL
+        },
+        .pin_source = GPIO_PinSource5,
+    },
+    .slave_count = 1,
+    .ssel = { {
+        .gpio = GPIOC,
+        .init = {
+            .GPIO_Pin = GPIO_Pin_14,
+            .GPIO_Speed = GPIO_Speed_50MHz,
+            .GPIO_Mode  = GPIO_Mode_OUT,
+            .GPIO_OType = GPIO_OType_PP,
+            .GPIO_PuPd = GPIO_PuPd_UP
+        },
+    } },
+};
+
+static uint32_t pios_spi_gyro_id;
+void PIOS_SPI_gyro_irq_handler(void)
+{
+    /* Call into the generic code to handle the IRQ for this specific device */
+    PIOS_SPI_IRQ_Handler(pios_spi_gyro_id);
+}
+#endif
+
+#if defined(PIOS_INCLUDE_FLASH)
+#include "pios_flashfs_logfs_priv.h"
+
+static const struct flashfs_logfs_cfg flashfs_internal_settings_cfg = {
+	.fs_magic      = 0x9ae1ee11,
+	.arena_size    = 0x00002000,       /* 32 * slot size = 8K bytes = 4 sectors */
+	.slot_size     = 0x00000100,       /* 256 bytes */
+};
+
+static const struct flashfs_logfs_cfg flashfs_internal_waypoints_cfg = {
+	.fs_magic      = 0x9ab4ee11,
+	.arena_size    = 0x00002000,       /* 32 * slot size = 8K bytes = 4 sectors */
+	.slot_size     = 0x00000100,       /* 256 bytes */
+};
+
+#include "pios_flash_internal_priv.h"
+
+static const struct pios_flash_internal_cfg flash_internal_cfg = {
+};
+
+#include "pios_flash_priv.h"
+
+static const struct pios_flash_sector_range stm32f3_sectors[] = {
+	{
+		.base_sector = 0,
+		.last_sector = 127,
+		.sector_size = FLASH_SECTOR_2KB,
+	},
+};
+
+uintptr_t pios_internal_flash_id;
+static const struct pios_flash_chip pios_flash_chip_internal = {
+	.driver        = &pios_internal_flash_driver,
+	.chip_id       = &pios_internal_flash_id,
+	.page_size     = 16, /* 128-bit rows */
+	.sector_blocks = stm32f3_sectors,
+	.num_blocks    = NELEMENTS(stm32f3_sectors),
+};
+
+static const struct pios_flash_partition pios_flash_partition_table[] = {
+	{
+		.label        = FLASH_PARTITION_LABEL_BL,
+		.chip_desc    = &pios_flash_chip_internal,
+		.first_sector = 0,
+		.last_sector  = 7,
+		.chip_offset  = 0,
+		.size         = (7 - 0 + 1) * FLASH_SECTOR_2KB,
+	},
+
+	{
+		.label        = FLASH_PARTITION_LABEL_SETTINGS,
+		.chip_desc    = &pios_flash_chip_internal,
+		.first_sector = 8,
+		.last_sector  = 15,
+		.chip_offset  = (8 * FLASH_SECTOR_2KB),
+		.size         = (15 - 8 + 1) * FLASH_SECTOR_2KB,
```

Comment:
This seems quite small for settings. Having more than 2 arenas may be good for wear-leveling (the STM32 internal flash has fewer erase/write cycles than typical SPI flash chips).